### PR TITLE
Added Impact to the use cases to justify requirements.

### PR DESCRIPTION
**Pull request by sridharangirish91**
Created a section to define all requirements and link them in the use cases for easier read.
Reworded a few things

### CODE REVIEW

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 1]
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
...
+: There might be many channels to signal the metadata such as
+(non-exhaustive list):
+
+  * TCP options {{?RFC9293}}
+  * UDP Options {{?I-D.ietf-tsvwg-udp-options}}
+  * IPv6 Hop-by-Hop Options ({{Section 4.3 of ?RFC8200}})
+  * QUIC CID mapping
+  * ICMP messages
+
+## System-Considerations
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
## System Considerations
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 2]
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
...
+
+REQ-PRIVACY-ADDITIONAL:
+: An on-path observer obtains no additional information about the IP
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
: An on-path observer obtains (or gleans) no additional information about the IP
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 3]
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
...
+packet.
+
+REQ-SIGNALING-AVOIDANCE:
+: Leveraging previous experience {{?RFC9049}}, the folliwing is not
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
: Leveraging previous experience {{?RFC9049}}, the following is not
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 4]
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
...
+required to make use of the collaborative signaling:
+
+  * Reveal the application identity.
+  * Expose the application cause (or 'reason') to signal metadata.
+  * Reveal server identity.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
  * Reveal the server identity.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 5]
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
...
+  * Inspect client-to-server encrypted payload by network elements.
+
+REQ-ISP-SCALE:
+: The metadata other state information that a wireless router has to
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
: The metadata and any other related state information that a network element (router) has to
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 6]
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
...
+maintain for each additional media flow it handles should be very low
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
maintain for each additional flow it handles should be very low
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 7]
```diff
@@ -349,8 +342,73 @@ network elements may drop loss tolerant packets during reactive
...
+: The network/OS needs to ensure that the user/client signaling of
+priority (if any) does not associate the same priority level with all
+traffic types within the same flow, thereby avoiding prioritizing of
+all the streams/traffic the same way.
 
+REQ-CLIENT-VALIDATION:
+: The network needs to ensure the signal is coming from the same
+user/client that is part of the 5-tuple flow.  This is to ensure no
+other application influences the priority of another application's
+flow.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
flow from within the same host.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 8]
```diff
@@ -363,27 +421,48 @@ Examples:
 
 1. Audio is more critical than video for many applications and should
 be prioritized differently than video.
-The client may indicate this preference to the network by selecting the importance value of audio packets as the highest priority.<br>
-<br>
-Requirement: REQ-PAYLOAD-CLIENT-DECIDES.
+The client may indicate this preference to the network by selecting
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
The client may indicate this preference to a network by selecting
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 9]
```diff
@@ -363,27 +421,48 @@ Examples:
...
+the importance value of audio packets as the highest priority.
+
+    Requirement: REQ-PACKET-PRIORITY, REQ-MEDIA-AV-SEPARATE,
+    REQ-PAYLOAD-CLIENT-DECIDES (defined in {{x-req-def}}).
+
+    Impact: With the above requirement met, audio clarity could be maintained
+  better in challenging network and during reactive events ensuring better user experience.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
  better in resource-constrained networks and during reactive events ensuring better user experience.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 10]
```diff
@@ -363,27 +421,48 @@ Examples:
...
 
 2. The server (or relay) sends the same stream to many receivers,
 including the same metadata (especially with media over QUIC).
-Some clients prefer video over audio and others audio over video based on local network conditions.
-This results in priority inversion in some cases.<br>
-<br>
-Requirement: REQ-PAYLOAD-CLIENT-DECIDES.
+Different clients have different priorities for different types of traffic. This would result
+in change in priorities for the same type of traffic a single server sends, based on the user/client.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
in change in priorities for the same type of traffic that a single server sends, based on the user/client.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 11]
```diff
@@ -363,27 +421,48 @@ Examples:
...
+
+    Requirement: REQ-PAYLOAD-CLIENT-DECIDES.
+
+    Impact: With the above requirement met, each client/user preferences are
+    prioritized accordingly while maintaining scalability on the server, since
+    the metadata the server sends still remains the same for all the connections.
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
    the metadata that the server sends still remains the same for all the connections.
```

File: draft-kwbdgrr-tsvwg-net-collab-rqmts.md [hunk 12]
```diff
@@ -363,27 +421,48 @@ Examples:
...
+
+4. In loss-prone networks or during Reactive Management events, if all packets of an application
+flow (UDP 4-tuple) such as live broadcast or on-demand video streaming are treated the same,
+it limits the ability to maximize network utilization and use the transiently available bandwidth.
+Dropping or delaying of (media) packets randomly is likely to lower network utilization
+and application performance.
+
+    Requirement: REQ-PACKET-PRIORITY.
+
+    Impact: By identifying packets that can be dropped, congestion can be reduced leading
```
Comment (boucadair) [inline draft-kwbdgrr-tsvwg-net-collab-rqmts.md]:
```suggestion
    Impact: By identifying packets that tolerate being dropped, congestion can be reduced leading
```

**Review by boucadair — CHANGES_REQUESTED** [review]

**Comment by boucadair** [comment]
Thanks.

For "REQ-METADATA-ACCURACY", please see my comment to Dan at #86.

**Comment by John-Kaippallimalil** [comment]
The main body of the draft now refers to 3 requirements (REQ-MEDIA-AV-SEPARATE; REQ-PAYLOAD-CLIENT-DECIDES, REQ-MEDIA-KEYFRAME) that are in the Appendix. This blurs the line on what sections in the Appendix are normative and informative. As a partial remedy, I have made 2 updates  (see my commits above) to indicate that these requirements are "enhanced".

**Comment by sridharangirish91** [comment]
> The main body of the draft now refers to 3 requirements (REQ-MEDIA-AV-SEPARATE; REQ-PAYLOAD-CLIENT-DECIDES, REQ-MEDIA-KEYFRAME) that are in the Appendix. This blurs the line on what sections in the Appendix are normative and informative. As a partial remedy, I have made 2 updates (see my commits above) to indicate that these requirements are "enhanced".

Hi John, thanks for the text. I agree with REQ-MEDIA-AV-SEPARATE and REQ-MEDIA-KEYFRAME but I doubt if REQ-PAYLOAD-CLIENT-DECIDES causes any issue with traffic analysis. I have moved the statements you have made into the requirements definition itself to keep the main body short and succinct while keeping the remedy in place as well.

**Comment by John-Kaippallimalil** [comment]
Hi Sridhar, Re: traffic analysis comment may apply for QUIC - where a the sender may schedule packets in any order (RR, priority, deadline, or even mix multiple media types in different FRAMES of the same packet). Plus there is the option to add chaff packets or stuff packets to not be able to detect based on size/frequency. This enhanced metadata should obviously not be used where effort is being taken to mask it from middleboxes in the network. On the other hand RTP is different (except if it is RoQ).

Having this kind of dependency between Appendix and main body is not ideal in my view. The gist of the impact/effect on other parts of the main body should be described there, as otherwise its hard for a reader to figure out what the impact would be to realize that requirement. Its fine if the preference is to to keep more of the details in the Appendix.

**Comment by danwing** [comment]
> Hi Sridhar, Re: traffic analysis comment may apply for QUIC - where a the sender may schedule packets in any order (RR, priority, deadline, or even mix multiple media types in different FRAMES of the same packet). Plus there is the option to add chaff packets or stuff packets to not be able to detect based on size/frequency. This enhanced metadata should obviously not be used where effort is being taken to mask it from middleboxes in the network. On the other hand RTP is different (except if it is RoQ).

Seems best to discuss once in Security Considerations (or Privacy Considerations) section, which we already discuss at https://github.com/SpencerDawkins/net-collab-rqmts/blob/main/draft-kwbdgrr-tsvwg-net-collab-rqmts.md#privacy-considerations-privacy

**Comment by sridharangirish91** [comment]
> The main body of the draft now refers to 3 requirements (REQ-MEDIA-AV-SEPARATE; REQ-PAYLOAD-CLIENT-DECIDES, REQ-MEDIA-KEYFRAME) that are in the Appendix. This blurs the line on what sections in the Appendix are normative and informative. As a partial remedy, I have made 2 updates (see my commits above) to indicate that these requirements are "enhanced".

If that's the case, then we can mention media streaming in extended use case as well, where we go little deeper into the use case and explain the extended requirements there while keeping the basic ones in the main section of the draft.

We can combine 1 and 3 in media streaming as just a way to prioritize different traffic differently and least priority ones can tolerate loss. We can then move 2 and add a use case where we can get into audio video separately as well as key frame there.
Alternatively, we could move CLIENT-PAYLOAD-DECIDES to the main doc and keep 2 there. Given AV and KeyFrame requirements are not required for Client-payload-decides and doesn't have any traffic analysis complications, it could be a valuable client-to-network signal in the main doc too.

**Comment by sridharangirish91** [comment]
@John-Kaippallimalil @danwing - Updated the text in the latest commit to keep extended use cases and requirements contained in the appendix.

**Review by John-Kaippallimalil — APPROVED** [review]
HI Sridhar,
This revision looks great. Thanks for the updates.

**Review by danwing — APPROVED** [review]

**Review by John-Kaippallimalil — APPROVED** [review]
Looks good.

**Review by boucadair — APPROVED** [review]

**Pull request merged**